src/cms/i18n.ts

```diff
@@ -0,0 +1,78 @@
+import * as R from 'ramda'
+
+import { Locale } from './i18n-locales'
+
+export const setLocales = <T extends Locale>(locales: T[]) => locales
+
+export interface LocalizedString {
+  locale: string
+  value: string
+}
+
+export type TranslationCollection = ReadonlyArray<LocalizedString>
+
+const isLocalizedString = (obj: any): obj is LocalizedString =>
+  R.allPass([
+    R.is(Object),
+    R.has('locale'),
+    R.has('value'),
+    R.pipe(R.keys, R.length, R.equals(2)),
+  ])(obj)
+
+export const isTranslationCollection = (
+  obj: any,
+): obj is TranslationCollection =>
+  R.allPass([
+    R.is(Array),
+    R.compose(R.not, R.isEmpty),
+    R.all(isLocalizedString),
+  ])(obj)
+
+const newTranslation = (locale: Locale, value: string): LocalizedString => ({
+  locale,
+  value,
+})
+
+export const uppendTranslation = (locale: Locale, value: string) => (
+  collection: TranslationCollection,
+): TranslationCollection =>
+  R.pipe(
+    R.reject<LocalizedString, 'array'>(R.propEq('locale', locale)),
+    R.ifElse(
+      R.pipe(R.always(value), R.isEmpty),
+      R.identity,
+      R.append(newTranslation(locale, value)),
+    ),
+  )(collection)
+
+// TODO: Use maybe functor to act on undefined values
+export const getTranslation = (locale: Locale) => (
+  collection: TranslationCollection,
+): string | undefined =>
+  R.pipe(
+    R.find(R.propEq('locale', locale)),
+    R.ifElse(R.isNil, R.identity, R.prop('value')),
+  )(collection)
+
+export const findTranslation = (locales: Locale[]) => (
+  collection: TranslationCollection,
+): string | undefined =>
+  R.pipe(
+    R.map((locale: Locale) => getTranslation(locale)(collection)),
+    R.reject(R.isNil),
+    R.head,
+  )(locales)
+
+export const translate = (...locales: Locale[]) => (node: any): any =>
+  R.cond([
+    [isTranslationCollection, findTranslation(locales)],
+    [
+      R.either(R.is(Array), R.is(Object)),
+      R.map(node => translate(...locales)(node)),
+    ],
+    [R.T, R.identity],
+  ])(node)
+
+export const createTranslateWithFallback = <T extends Locale>(
```

Comment:
I think in most (maybe: all) circumstances the fallbacks array will be the LOCALES constant from i18n.config.ts. So maybe add this as a default argument here
```suggestion
export const createTranslate = <T extends Locale>(  fallbacks: T[] = LOCALES) => (locale: T) => translate(...R.prepend(locale, fallbacks))
```

Reply:
Perhaps also add it as a default value in all the other functions (translate, findTranslation)

Reply:
Locales are set in the `i18n.ts` file!